### PR TITLE
Auto capture msg

### PR DESCRIPTION
**Pull request by summeroff**
this branch include changes from another auto game capture PR: 
https://github.com/stream-labs/obs-studio/pull/170

### CODE REVIEW

File: plugins/win-capture/game-capture.c [hunk 1]
```diff
@@ -592,21 +601,104 @@ static void load_placeholder_image(struct game_capture *gc)
...
+		translated_string = malloc( (len+1)*2);
+		os_utf8_to_wcs(gc->placeholder_text.array, 0, &translated_string[0], len + 1);
+	} else {
+		return;
+	}
+
+	struct obs_video_info ovi;	
+	obs_get_video_info(&ovi);
+
+	gc->placeholder_text_height = ovi.base_height/8;
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c]: Why do we divide by 8?

File: plugins/win-capture/game-capture.c [hunk 2]
```diff
@@ -592,21 +601,104 @@ static void load_placeholder_image(struct game_capture *gc)
...
+	gc->placeholder_text_width = ovi.base_width;
+
+	BITMAPINFOHEADER bmphdr = { 0 };
+	bmphdr.biSize = sizeof(BITMAPINFOHEADER);
+	bmphdr.biWidth = gc->placeholder_text_width;
+	bmphdr.biHeight = -gc->placeholder_text_height;
+	bmphdr.biPlanes = 1;
+	bmphdr.biBitCount = 32;
+	bmphdr.biSizeImage = gc->placeholder_text_height * gc->placeholder_text_width * 4;
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c]: I assume we multiple by 4 because the format is RGBA and it's 4 bytes per pixel? If so can put this 4 into a constant variable with a meaningful name?

File: plugins/win-capture/game-capture.c [hunk 3]
```diff
@@ -592,21 +601,104 @@ static void load_placeholder_image(struct game_capture *gc)
...
+	
+	HDC wdc = GetDC(NULL);
+	if (wdc) {
+		HDC memDC = CreateCompatibleDC ( wdc );
+
+		uint8_t * mdata = bmalloc(gc->placeholder_text_height*gc->placeholder_text_width*4);
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c]: Same comment about the multiplication by 4 here, since we need the size at multiple places, perhaps it would be better to put it in a single variable.

File: plugins/win-capture/game-capture.c [hunk 4]
```diff
@@ -592,21 +601,104 @@ static void load_placeholder_image(struct game_capture *gc)
...
+			while (true) {
+				font = CreateFontIndirect(&LogFont);
+				SelectObject(memDC, font);
+				
+				GetTextExtentPoint32(memDC, translated_string, len, &string_width);
+				if (string_width < gc->placeholder_text_width) 
+					break;
+				
+				DeleteObject(font);
+				LogFont.lfHeight = LogFont.lfHeight - 5;
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c]: why are we substracting 5? Not clear, please use a meaning full constant variable.

File: plugins/win-capture/game-capture.c [hunk 5]
```diff
@@ -592,21 +601,104 @@ static void load_placeholder_image(struct game_capture *gc)
...
+			}
+
+			TextOut(memDC, (gc->placeholder_text_width-string_width)/2, (gc->placeholder_text_height-LogFont.lfHeight)/2, translated_string, len); 
+
+			for( int i = 0; i <gc->placeholder_text_height*gc->placeholder_text_width; i++)	{
+				int colores = mdata[i*4+0] + mdata[i*4+1] + mdata[i*4+2];
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c]: Very hard to follow this calculation, please use constant variables.

File: plugins/win-capture/game-capture.c [hunk 6]
```diff
@@ -638,12 +730,14 @@ static void game_capture_update(void *data, obs_data_t *settings)
...
-	{
+	const char *img_path = obs_data_get_string(settings, SETTING_PLACEHOLDER_IMG);
+	if(gc->placeholder_image_path.len == 0 || dstr_cmp(&gc->placeholder_image_path, img_path) != 0 ) {
 		unload_placeholder_image(gc);
 	}
- 	dstr_copy(&gc->placeholder_img, placeholder_img);
+ 	dstr_copy(&gc->placeholder_image_path, img_path);
+
+	const char *placeholder_text = obs_data_get_string(settings, SETTING_PLACEHOLDER_MSG);
+    dstr_copy(&gc->placeholder_text, placeholder_text);
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c]: Style issue.

File: plugins/win-capture/game-capture.c [hunk 7]
```diff
@@ -592,21 +601,109 @@ static void load_placeholder_image(struct game_capture *gc)
...
+			TextOut(memDC, (gc->placeholder_text_width-string_width)/2, (gc->placeholder_text_height-LogFont.lfHeight)/2, translated_string, len); 
+
+			for( int i = 0; i <gc->placeholder_text_height*gc->placeholder_text_width; i++)	{
+				int pixel_offset = i*bytes_per_pixel;
+				int text_pixel_color_components_average = (text_bitmap_buffer[pixel_offset+0] + text_bitmap_buffer[pixel_offset+1] + text_bitmap_buffer[pixel_offset+2])/3;
+				text_bitmap_buffer[pixel_offset + ALPHA_COMPONENT] = text_pixel_color_components_average;
+			}
+
+			obs_enter_graphics();
+			gc->placeholder_text_texture = gs_texture_create(gc->placeholder_text_width, gc->placeholder_text_height, GS_BGRA, 1, &text_bitmap_buffer, GS_DYNAMIC);
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c]: Styling issues, from line 674 to 683 -> very hard to read, please add line return to ease the reading.

**Comment by summeroff** [comment]
This PR have some changes from another game capture branch.
Cleaned up PR resubmitted as 
https://github.com/stream-labs/obs-studio/pull/191

**Comment by EddyGharbi** [comment]
This PR needs to be rebased to the current base branch.
